### ISSUE TITLE
Running `cou plan --parallel` do not make sense

### ISSUE THREAD

**Issue by rgildein** [canonical/charmed-openstack-upgrader#110]
The `--parallel` option is available for the `plan' command, which makes no sense.

```bash
$ cou plan --help            
Usage: cou plan [options]

Show the steps COU will take to upgrade the cloud to the next release.

Options:
  -h, --help          Show this help message and exit.
  --model MODEL_NAME  Set the model to operate on.
                      If unset, the model name will be determined by inspecting the environment as follows:
                        1 - Environment variable JUJU_MODEL,  2 - Environment variable MODEL_NAME,  3 - Current active juju model
  --parallel          Run upgrade steps in parallel where possible.
  --verbose, -v       Increase logging verbosity in STDOUT. Multiple 'v's yield progressively more detail (up to 4).
                      Note that this doesn't affect the verbosity in logfile, which will always have the maximum verbosity.
  --quiet, -q         Disable output in STDOUT. 
```

Even the description is suggesting that this will be used while running upgrade steps.

**Comment by agileshaw** [comment]
I don't agree that `cou plan --parallel` doesn't make sense because in data-plane stage the plan COU generates should contain the information on how the parallelization groups are formed when passing the flag. But, as we concluded in earlier discussion today, we are going to remove `--parallel` option all together and make parallel execution a default, so plan should contain that grouping information by default as well.

**Comment by rgildein** [comment]
I think that even if we left the parameter `--parallel` that should not change any results from Analyze or plan, since it's relevant only for running upgrade.

**Comment by agileshaw** [comment]
It is relevant to the plan phase because we can either plan for serial upgrade or parallel upgrade (with different upgrade step layout), especially with data-plane components. But since we are going to make parallel the default, this discussion is out-of-scope here. We can talk about UI design for plan result when we are ready for data-plane upgrades.
